packages/react-native-codegen/src/parsers/parsers-commons.js

```diff
@@ -432,6 +439,76 @@ function buildSchema(
   );
 }
 
+const parseModuleName = (
+  hasteModuleName: string,
+  moduleSpec: $FlowFixMe,
+  ast: $FlowFixMe,
+  parser: Parser,
+): string => {
+  const callExpressions = [];
+  visit(ast, {
+    CallExpression(node) {
+      if (isModuleRegistryCall(node)) {
+        callExpressions.push(node);
+      }
+    },
+  });
+
+  throwIfUnusedModuleInterfaceParserError(
+    hasteModuleName,
+    moduleSpec,
+    callExpressions,
+  );
+
+  throwIfMoreThanOneModuleRegistryCalls(
+    hasteModuleName,
+    callExpressions,
+    callExpressions.length,
+  );
+
+  const [callExpression] = callExpressions;
+  let typeParameters;
+  if (parser.language() === 'TypeScript') {
+    typeParameters = callExpression.typeParameters;
+  } else {
+    typeParameters = callExpression.typeArguments;
+  }
```

Comment:
can we create a function `callExpressionTypeParameters` in the `Parser` and implement it in the FlowParser so that it returns `callExpression.typeArguments;` and in the TypeScriptParser so that it returns `callExpression.typeParameters;`?

In this way, we can replace all these lines with just:
```js
const typeParameters = parser.callExpressionTypeParameters(callExpression);
```

Reply:
Definitely. That should also simplify similar checks at other places.

Reply:
> I think that the only part missing is to test [this check](https://github.com/facebook/react-native/blob/main/packages/react-native-codegen/src/parsers/flow/parser.js#L72). So if the TypeParameter is not a GenericTypeAnnotation, it should throw.

@cipolleschi as mentioned by me [here](https://github.com/facebook/react-native/pull/36297#discussion_r1118061884) ->
>  @cipolleschi
I've tried various ways to test this error, but I couldn't find a way to make it work.
Since I'm not yet; too familiar with parsers' functionalities, syntax and use cases, I don't want to spend more time on this.
I've left the commented code below as a reference for the future.
If you or anyone know how to test this error properly, please feel free to add/complete the tests.
Thanks!

I know we need to [test this](https://github.com/facebook/react-native/blob/main/packages/react-native-codegen/src/parsers/flow/parser.js#L72) but I couldn't get it to work after trying so many times. I know it must be an easy and typical test to write but, I'm just unable to :( ...

Will you suggest code for the **throw** case when you get the time?